common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -1718,6 +1698,32 @@ private MsalOAuth2TokenCache getTokenCacheForClient(@NonNull final String client
         return targetCache;
     }
 
+    /**
+     * Returns the {@link MsalOAuth2TokenCache} to use for supplied client id and environment.
+     *
+     * @param clientId          The target client id.
+     * @param environment       The target environment.
+     * @param callingProcessUid The uid of the calling process.
```

Comment:
Probably worth calling out somewhere that this identifier is unique and assigned at installation time.  With that said I don't think anything stops these identifiers from being recycled.

Reply:
b70689bf406336908d169895bc1f7408db842c96